src/Components/InMemoryEventStream.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Components
+{
+    internal sealed class InMemoryEventStream : EventStream, EventDispatcher
+    {
+        private readonly HashSet<object> _subscriptions = new HashSet<object>();
+
+        public void Dispatch<T>(T entity)
+        {
+            if (entity == null)
+                throw new ArgumentNullException(nameof(entity));
+
+            lock(_subscriptions)
```

Comment:
Are concurrent dictionary/queue/stack the appropriate data structure for this? nope. Can you remove a single **known** item from a concurrent bag, nope! Partioners are out as well as the blocking collection as far as I can tell. This leaves us with the standard non-concurrent ones. Hashset<T> is probably the next best + locks for safety. Herecy pointed out that we possibly dont need the lock but knowing how funky List<T> is while enumerating, i'd be err'ing on the side of caution till we have tested it. :)